src/Analysis/Core/Impl/Interpreter/PythonLibraryPath.cs

```diff
@@ -109,31 +109,30 @@ public sealed class PythonLibraryPath : IEquatable<PythonLibraryPath> {
         /// <param name="library">Root of the standard library.</param>
         /// <returns>A list of search paths for the interpreter.</returns>
         /// <remarks>New in 2.2, moved in 3.3</remarks>
-        private static List<PythonLibraryPath> GetDefaultSearchPaths(IFileSystem fs, string library) {
-            var result = new List<PythonLibraryPath>();
+        private static ImmutableArray<PythonLibraryPath> GetDefaultSearchPaths(IFileSystem fs, string library) {
```

Comment:
Not entirely related to this PR, but I feel like we should come up with some agreement about which list types we're going to use across the board so we don't end up switching them back and forth between PRs...

Reply:
I am OK either way but I tend to use interfaces on public surface. Is there a benefit of `Immutable` over `IReadOnly`?

Reply:
There are 3 issues with `IReadOnlyList<T>`:
1. It is read-only, but not immutable. If an object is provided to the outer context, but owner still has a reference to the object, the is no way to synchronize it anymore. E.g.:
    ```cs
    private List<SomeType> _someList;
    public IReadOnlyList<SomeType> SomeList=> _someList;
    ```
    If any code gets reference to the list through `SomeList`, it still can be changed by the owner, concurrently to the reader.
2. Adding elements. With `ImmutableArray`, adding elements create new instance of it, but it is as cheap as adding to `List<T>`. For `IReadOnlyList<T>`, it requires to create a copy of the list.
3. Enumerator. For any interface, enumerator is an object. For `ImmutableArray<T>` and `List<T>` it would be a struct. This becomes a problem when we have `foreach` loops in analysis where enumerators are promoted to Gen2.